From 6790dbf174c1462387daa5b59943e9e709393023 Mon Sep 17 00:00:00 2001
From: Michael Pontus <m.pontus@gmail.com>
Date: Wed, 31 Aug 2022 06:54:09 +0000
Subject: [PATCH 1/4] disable sha512 verification

---
 contrib/verify-commits/verify-commits.py | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/contrib/verify-commits/verify-commits.py b/contrib/verify-commits/verify-commits.py
index 2ff14c1..71c12ac 100755
--- a/contrib/verify-commits/verify-commits.py
+++ b/contrib/verify-commits/verify-commits.py
@@ -134,13 +134,6 @@ def main():
                 print("{} was not signed with a trusted key!".format(current_commit), file=sys.stderr)
             sys.exit(1)
 
-        # Check the Tree-SHA512
-        if (verify_tree or prev_commit == "") and current_commit not in incorrect_sha512_allowed:
-            tree_hash = tree_sha512sum(current_commit)
-            if ("Tree-SHA512: {}".format(tree_hash)) not in subprocess.check_output([GIT, 'show', '-s', '--format=format:%B', current_commit]).decode('utf8').splitlines():
-                print("Tree-SHA512 did not match for commit " + current_commit, file=sys.stderr)
-                sys.exit(1)
-
         # Merge commits should only have two parents
         parents = subprocess.check_output([GIT, 'show', '-s', '--format=format:%P', current_commit]).decode('utf8').splitlines()[0].split(' ')
         if len(parents) > 2:
-- 
2.37.1

