From 9b703be2ab2365e41aa850906c454044d5f7ba72 Mon Sep 17 00:00:00 2001
From: Michael Pontus <m.pontus@gmail.com>
Date: Wed, 31 Aug 2022 09:09:32 +0000
Subject: [PATCH 4/4] relative paths

---
 contrib/verify-commits/gpg.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/contrib/verify-commits/gpg.sh b/contrib/verify-commits/gpg.sh
index 1d36277..0c4e210 100755
--- a/contrib/verify-commits/gpg.sh
+++ b/contrib/verify-commits/gpg.sh
@@ -40,7 +40,7 @@ for LINE in $GPG_RES; do
 	"[GNUPG:] VALIDSIG "*)
 		while read KEY; do
 			[ "${LINE#?GNUPG:? VALIDSIG * * * * * * * * * }" = "$KEY" ] && VALID=true
-		done < ./contrib/verify-commits/trusted-keys
+		done < "$(dirname $0)/trusted-keys"
 		;;
 	"[GNUPG:] REVKEYSIG "*)
 		[ "$BITCOIN_VERIFY_COMMITS_ALLOW_REVSIG" != 1 ] && exit 1
-- 
2.37.1

